### PR TITLE
Feature/EUI 8139

### PR DESCRIPTION
**Pull request by DavidJayakumar**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/EUI-8139

### Change description ###

Add sorting for QM table

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/query-management/components/query-list/query-list.component.ts [hunk 1]
```diff
@@ -1,18 +1,90 @@
...
+        this.sort(col);
+        break;
+      }
+    }
+  }
 
-  constructor() {
+  public sortWidget(col: column): string {
+    switch (col.sortOrder) {
+      case SortOrder.DESCENDING: {
```
Comment (MateiHMCTS) [inline projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/query-management/components/query-list/query-list.component.ts:55]: I think the order should be different here;

DESCENDING should be a down triangle
ASCENDING should be an up triangle (smaller at the top and increasing as it goes to bottom).

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/query-management/components/query-list/query-list.component.ts [hunk 2]
```diff
@@ -9,11 +10,81 @@ import { PartyMessagesGroup, QueryListData } from '../../domain';
...
+      }
+      case 'Last response date': {
+        this.sortDate(col);
+        break;
+      }
+      case 'Response by': {
+        this.sort(col);
+        break;
+      }
+    }
```
Comment (Daniel-Lam) [inline projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/query-management/components/query-list/query-list.component.ts:50]: @DavidJayakumar - minor point, the linting might pick up that there's no `default` clause... might be worth adding one

**Review by Daniel-Lam — CHANGES_REQUESTED** [review]
@DavidJayakumar - just one minor observation; the rest is OK

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/query-management/components/query-list/query-list.component.ts (hunk 1 above)
Comment (DavidJayakumar) [reply projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/query-management/components/query-list/query-list.component.ts:59]: changed

**Review by Daniel-Lam — DISMISSED** [review]

**Review by MateiHMCTS — APPROVED** [review]

**Pull request merged**